### PR TITLE
objective descriptions with icon replacement

### PR DESCRIPTION
**Pull request by sundevour**
re-up of #3935 because a bad merge turned it into spaghetti

most review items done, did not solve the removal of `.slice`

please review
- whether `_.once` is implemented better now
- does this need an errorboundary somewhere?

### CODE REVIEW

File: src/app/progress/ObjectiveDescription.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  // check remainingObjectiveString for replaceable strings
+  const matchResults = remainingObjectiveString.match(hashExtract);
+
+  // return immediately if there's nothing to try and replace
+  if (!matchResults) {
+    return alreadyProcessed.concat([remainingObjectiveString]);
+  }
+
+  // set variables to do replacement
+  const [beforeMatch, iconString, afterMatch] = matchResults.slice(1);
```
Comment (delphiactual) [inline src/app/progress/ObjectiveDescription.tsx]: `const [, beforeMatch, iconString, afterMatch] = matchResults;`

File: src/app/progress/ObjectiveDescription.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    const progressDescription = objectiveDef.progressDescription;
+
+    // match the first bracketed item, or the first zh character, plus beforestuff and afterstuff
+    const iconString = progressDescription.match(hashExtract)[2];
+
+    // the identified iconString is the manifest's replacement marker for this icon. put back into table
+    baseConversionTable[index].substring = iconString;
+  });
+
+  console.log(baseConversionTable);
```
Comment (delphiactual) [inline src/app/progress/ObjectiveDescription.tsx]: remove

File: src/app/progress/ObjectiveDescription.tsx (hunk 1 above)
Comment (sundevour) [reply src/app/progress/ObjectiveDescription.tsx]: niiiiiice

File: src/app/progress/ObjectiveDescription.tsx (hunk 2 above)
Comment (sundevour) [reply src/app/progress/ObjectiveDescription.tsx]: ty! ✅

**Pull request merged**